pkg/identityserver/registry_search.go

```diff
@@ -446,6 +446,47 @@ func (rs *registrySearch) SearchUsers(ctx context.Context, req *ttnpb.SearchUser
 	return res, nil
 }
 
+func (rs *registrySearch) SearchAccounts(
+	ctx context.Context, req *ttnpb.SearchAccountsRequest,
+) (*ttnpb.SearchAccountsResponse, error) {
+	authInfo, err := rs.authInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Limit the amount of data we provide to non-admin users when searching across all accounts.
+	if req.CollaboratorOf == nil && !authInfo.IsAdmin {
+		// Require a search query of at least 3 characters.
+		if len(req.Query) < 3 {
+			return &ttnpb.SearchAccountsResponse{}, nil
+		}
```

Comment:
I get the rationale but this also means that users with 2 character IDs cannot be found.

Reply:
Fair point, updated this to require a minimum length of 2